migrations/V111__Remap_Existing_Model_Name_To_FK.sql

```diff
@@ -0,0 +1,51 @@
+-- Adding new columns to the plan_general_characteristics table
+ALTER TABLE plan_general_characteristics
+  ADD COLUMN existing_model_plan_id UUID NULL,
+  ADD COLUMN existing_model_external_id INT NULL;
+
+-- Adding foreign key constraints
+ALTER TABLE plan_general_characteristics
+  ADD CONSTRAINT fk_existing_model_plan
+    FOREIGN KEY (existing_model_plan_id)
+      REFERENCES model_plan(id)
+      ON DELETE SET NULL;
+
+ALTER TABLE plan_general_characteristics
+  ADD CONSTRAINT fk_existing_model_external
+    FOREIGN KEY (existing_model_external_id)
+      REFERENCES public.existing_model(id)
+      ON DELETE SET NULL;
+
+-- Ensuring that existing_model_plan_id and existing_model_external_id are mutually exclusive
+ALTER TABLE plan_general_characteristics
+  ADD CONSTRAINT plan_general_characteristics_check_exclusive_model_ids CHECK (
+      (existing_model_plan_id IS NULL) OR (existing_model_external_id IS NULL)
+    );
+
+-- Data migration for existing_model_plan_id
+UPDATE plan_general_characteristics pc
+SET existing_model_plan_id = (
+  SELECT mp.id
+  FROM model_plan mp
+  WHERE mp.model_name = pc.existing_model
+  LIMIT 1
+)
+WHERE EXISTS (

```

Comment:
I think this could make more sense as a join rather than two sub-queries. Ultimately, if it functions it doesn't matter too much since this will be run only once. I'd recommend taking a look at migration V91 if you are curious about structuring it with joins. You can also update for both existing and current model plan with one query that way, and also compare if they match to both instead of just one
```SQL
    LEFT JOIN existing_model ON pc.existing_model = existing_model.model_name
    LEFT JOIN model_plan ON pc.existing_model = model_plan.model_name
```